cmd/export_ledgers_test.go

```diff
@@ -85,38 +79,48 @@ func TestTransformLedger(t *testing.T) {
 			golden:  "10_ledgers.golden",
 			wantErr: nil,
 		},
+		{
+			name:    "range too large",
+			args:    []string{"export_ledgers", "-s", "30822015", "-e", "30822025", "-l", "5", "-o", testFileName},
+			golden:  "large_range_ledgers.golden",
+			wantErr: nil,
+		},
 	}
 
 	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			dir, err := os.Getwd()
+		runCLITest(t, test, "testdata/ledgers/")
+	}
+}
+
+func runCLITest(t *testing.T, test cliTest, goldenFolder string) {
+	t.Run(test.name, func(t *testing.T) {
+		dir, err := os.Getwd()
+		assert.NoError(t, err)
+
+		cmd := exec.Command(path.Join(dir, executableName), test.args...)
+		errText, actualError := cmd.CombinedOutput()
+
+		// Since the CLI uses a logger to report errors, the final error message isn't the same as the errors thrown in code.
+		// Instead, it's wrapped in other os/system errors
+		// By reading the error text from the logger, we can extract the lower level error that the user would see
+		if test.golden == "" {
+			errorMsg := fmt.Errorf(extractErrorMsg(string(errText)))
+			assert.Equal(t, test.wantErr, errorMsg)
+		}
+
+		// Tests that are designed to fail won't generate an valid output file, and so testing shouldn't look for one
+		if test.golden != "" {
+			assert.Equal(t, test.wantErr, actualError)
+			actualOutput, err := ioutil.ReadFile(testFileName)
```

Comment:
Thinking more, there are two viable routes to testing the output.
1. What you're doing here: write to a file, read that file back in testing, check it against a reference.
2. Add a CLI flag to write the output to stdout, check that output in memory.

I think 2 is better for testing, since it avoids potential memory issues in a future CI system. For this PR, this is fine. I would just add a TODO to add a flag for an output option; implement stdout; and update tests accordingly.